src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Remove.cs

```diff
@@ -0,0 +1,579 @@
+                        {
+                            if (!DType.String.Accepts(argType, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules) &&
+                                !BuiltInEnums.RemoveFlagsEnum.FormulaType._type.Accepts(argTypes[i], exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules))
+                            {
+                                fValid = false;
+                                errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrRemoveAllArg);
+                            }
+                        }
+                        else
+                        {
+                            if (!BuiltInEnums.RemoveFlagsEnum.FormulaType._type.Accepts(argTypes[i], exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules))
+                            {
+                                fValid = false;
+                                errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrRemoveAllArg);
+                            }
+                        }
+
+                        continue;
+                    }
+                    else
+                    {
+                        fValid = false;
+                        errors.EnsureError(args[i], TexlStrings.ErrNeedRecord_Arg, args[i]);
+                        continue;
+                    }
+                }
+
+                var collectionAcceptsRecord = collectionType.Accepts(argType.ToTable(), exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules);
+                var recordAcceptsCollection = argType.ToTable().Accepts(collectionType, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: context.Features.PowerFxV1CompatibilityRules);
+
+                // The item schema should be compatible with the collection schema.
+                if (!collectionAcceptsRecord && !recordAcceptsCollection)
+                {
+                    fValid = false;
+                    if (!SetErrorForMismatchedColumns(collectionType, argType, args[i], errors, context.Features))
+                    { 
+                        errors.EnsureError(DocumentErrorSeverity.Severe, args[i], TexlStrings.ErrCollectionDoesNotAcceptThisType);
+                    }
+                }
+                
+                // Only warn about no-op record inputs if there are no data sources that would use reference identity for comparison.
+                else if (!collectionType.AssociatedDataSources.Any() && !recordAcceptsCollection)
+                {
+                    errors.EnsureError(DocumentErrorSeverity.Warning, args[i], TexlStrings.ErrCollectionDoesNotAcceptThisType);
+                }
+
+                if (!context.AnalysisMode)
```

Comment:
>AnalysisMode [](http://example.com/codeflow?start=29&length=12)

Like the previous comment, should this be based on AnalysisMode? Or PFxV1 would be more appropriate?